.github/workflows/tests.yaml

```diff
@@ -1,36 +1,50 @@
 name: Build
-on: [push]
+
+on:
```

Comment:
any reason to limit ourselves here?

Reply:
this is a common pattern for FOSS projects:
* only build the "release" branches (like `main`)
* when you or someone else pushes modification branches, there is little value in building those branches directly. Instead, you want to build the pull request **into** one of the release branches (main)
* this allows you and everyone else contributing to this project to work on the same system - me or you or anyone must create a PR for it to be built

Reply:
I mean, I have no objections, but we could also leave it as a free for all, it's not like we have millions of commits every day :-)

Reply:
??  it is free for all - my point is that you don't want to compile it 3 times:  on branch creation, on pull creation, and on merge to `main` -- simply because the first two are identical, and you should have PR before merging.

Reply:
P.S. note that if you modify the branch from which a PR was created, that triggers rebuild on both `branch` and `pull` -- so you get two builds on each change

Reply:
from my understanding, currently we run CI for every commit (i.e. free for all), and with this PR we'll only run for a subset of commits

The proposed way works for me, we're good

Reply:
not exactly - it runs when **you** commit anything to your repo.

* When I create a new PR, I first create a branch -- which trigger a local action run  (on my own fork). The results are local to my own fork.
* Afterwards, I create a PR, but you do NOT see the results of a build -- because the build happened on BRANCH before the PR was created (i.e. the build is not associated with the PR).
* Whenever I update the branch (e.g. per your feedback), the build once again happens locally on my fork - the PR does not show its results

So in short, it is far better to build on PR change rather than branch change - because it keeps all those build results in one place - associated with the PR

Reply:
ooooooh, gotcha, you're right, it's much better